src/wallet/rpcwallet.cpp

```diff
@@ -2937,7 +2937,6 @@ UniValue loadwallet(const JSONRPCRequest& request)
     if (!wallet) {
         throw JSONRPCError(RPC_WALLET_ERROR, "Wallet loading failed.");
     }
-    AddWallet(wallet);
```

Comment:
This feels VERY unrelated

Reply:
Part of https://github.com/dashpay/dash/pull/3682#discussion_r498753310, Removed it there because `AddWallet` is called already in `CreateWalletFromFile` with the related changes

Reply:
Are these changes a necessity to be in this PR? they feel like they should be broken out

Reply:
It will simply not work without those changes in here unless we refactor `interfaces::PrivateSend::Client` and/or move it out of `interfaces::Wallet`. But IMO its pretty nice the way they are cooperated right now and i would definitely vote to leave it as it is but im also always open for concrete suggestions.